src/redisraft.c

```diff
@@ -596,14 +586,18 @@ static void handleRedisCommandAppend(RedisRaftCtx *rr,
     }
 
     /* Confirm that we're the leader and handle redirect or proxying if not. */
-    Node **ret = rr->config->follower_proxy ? &leader_proxy : NULL;
+    Node **ret = rr->config->follower_proxy || cmds->asking ? &leader : NULL;
```

Comment:
  I suggest to change the `checkLeader` behaviour to check if we are the leader, or if ret exists fill it with value - without reply anything + having a wrapper method that based on the asking status or follower_proxy value reply :
!leader && asking -> reply ASK, 
!leader && !follower_proxy-> reply redirect)
I feel it's hard to follow the logic in the current implementation.